simple_history/utils.py

```diff
@@ -1,11 +1,184 @@
+import sys
+from contextlib import contextmanager
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Callable, ClassVar, Iterator, Optional, Type, Union
+
+from asgiref.local import Local
+from django.conf import settings
 from django.db import transaction
-from django.db.models import Case, ForeignKey, ManyToManyField, Q, When
+from django.db.models import Case, ForeignKey, ManyToManyField, Model, Q, When
 from django.forms.models import model_to_dict
 
-from simple_history.exceptions import AlternativeManagerError, NotHistoricalModelError
+from .exceptions import AlternativeManagerError, NotHistoricalModelError
+
+if TYPE_CHECKING:
+    from .manager import HistoricalQuerySet, HistoryManager
+    from .models import HistoricalChanges
+
+
+@contextmanager
+def disable_history(
+    *,
+    only_for_model: Type[Model] = None,
+    instance_predicate: Callable[[Model], bool] = None,
+) -> Iterator[None]:
+    """
+    Disable creating historical records while this context manager is active.
+
+    Note: ``only_for_model`` and ``instance_predicate`` cannot both be provided.
+
+    :param only_for_model: Only disable history creation for this model type.
+    :param instance_predicate: Only disable history creation for model instances passing
+        this predicate.
+    """
+    assert (  # nosec
+        _StoredDisableHistoryInfo.get() is None
+    ), "Nesting 'disable_history()' contexts is undefined behavior"
+
+    if only_for_model:
+        # Raise an error if it's not a history-tracked model
+        get_history_manager_for_model(only_for_model)
+        if instance_predicate:
+            raise ValueError(
+                "'only_for_model' and 'instance_predicate' cannot both be provided"
+            )
```

Comment:
I'm curious why you decided on this. I'm generally fine with it, but we could `AND` them together.